reconcile/gitlab_labeler.py

```diff
@@ -10,6 +11,60 @@
 QONTRACT_INTEGRATION = 'gitlab-labeler'
 
 
+def get_app_list() -> Dict:
+    """
+    Get a list of all services managed by app-interface with the parent app
+    and the onboarding status of the service.
+    """
+    apps = queries.get_apps()
+    app_info = {a['name']: {'onboardingStatus': a['onboardingStatus'],
+                'parentApp': a['parentApp']} for a in apps}
+    return app_info
+
+
+def get_parents_list() -> List[str]:
+    """
+    Get a set of all service names that are parents of another service
+    """
+    parent_set = set()
+    apps = queries.get_apps()
+    for a in apps:
+        if a['parentApp'] is not None:
+            parent_set.add(a['parentApp']['name'])
+
+    return list(parent_set)
+
+
+def guess_onboarding_status(changed_paths: Iterable[str],
+                            apps: Dict[str, Dict], parent_apps:
+                            Set[str]) -> Optional[str]:
+    """
+    Guess the service name of a given MR from the changed paths of the
+    MR. This will allow to add the onboarding status to the MR's as label
+    """
+    labels = set()
+    for path in changed_paths:
+        if 'data/services/' in path:
+            path_slices = path.split('/')
+            try:
+                if path_slices[2] in parent_apps:
+                    app = apps[path_slices[3]]
+                    if app is not None:
+                        labels.add(app['onboardingStatus'])
+                else:
+                    app = apps[path_slices[2]]
+                    if app is not None:
+                        labels.add(app['onboardingStatus'])
+                    else:
+                        return None
+            except (IndexError, KeyError):
+                logging.info("Error getting app name")
+                return None
+
+    labels_list = list(labels)
+    return labels_list[0] if len(labels_list) == 1 else None
```

Comment:
Easier:
```suggestion
    if len(labels) == 1:
        return labels.pop()
    else:
        return None
```

Reply:
there could be multiple apps involved in a MR, and thus multiple onboarding statuses?

Reply:
Addressed this in https://github.com/app-sre/qontract-reconcile/pull/2001#discussion_r760291999